pkg/autoscaler/autoscaler.go

```diff
@@ -199,25 +204,49 @@ func (agg *perPodAggregation) podWeight(now time.Time) float64 {
 // Autoscaler stores current state of an instance of an autoscaler
 type Autoscaler struct {
 	*DynamicConfig
-	key          string
-	target       float64
-	stats        map[statKey]Stat
-	statsMutex   sync.Mutex
-	panicking    bool
-	panicTime    *time.Time
-	maxPanicPods float64
-	reporter     StatsReporter
-	targetMutex  sync.RWMutex
+	key             string
+	namespace       string
+	revisionService string
+	endpointsLister corev1listers.EndpointsLister
+	panicking       bool
+	panicTime       *time.Time
+	maxPanicPods    float64
+	reporter        StatsReporter
+
+	// targetMutex guards the elements in the block below.
+	targetMutex sync.RWMutex
+	target      float64
+
+	// statsMutex guards the elements in the block below.
+	statsMutex sync.Mutex
+	stats      map[statKey]Stat
+
+	// readyPodsMutex guards the elements in the block below.
+	readyPodsMutex sync.RWMutex
+	readyPodsCache map[time.Time]float64
```

Comment:
I have the same question as markus' above but can't find where the response is - I cant tell the reasoning for this readyPodsCache as we are only reading it when we perform Scale so this shouldnt buy us anything over the existing Informer cache?

Reply:
It's partially ground-work for the upcoming scraping work where you'll have to have a backlog of past available ready pods. That's also why that cache is "cut" after 60 seconds.

I think we'll be able to improve on this design by for example cache the past computations vs. the bits you need to compute things but we can always refactor that later.

Reply:
Ah, interesting - thanks!

Reply:
`cache the past computations vs. the bits you need to compute things` makes sense to me. I removed the cache.